src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,41 @@ mod bar {
 ```
 "##,
 
+E0254: r##"
+This error indicates that an imported symbol resolved to an already imported
+symbol.
+
+An example of this error:
+
+```
+// crate foo
+pub use foo::bar;
+pub mod foo; // note that there's a `pub` here
```

Comment:
the `pub` is only relevant due to the use of a glob import ... my opinion is that it distracts from the real issue that we want to describe for this error code: Namely the conflict between one `extern crate` and some other import into the type+module namespace.